R/utils.R

```diff
@@ -47,6 +47,29 @@ include_parent_datanames <- function(dataname, join_keys) {
   return(unique(c(parents, dataname)))
 }
 
+
+
+#' Create a `FilteredData`
+#'
+#' Create a `FilteredData` object from a `teal_data` object
+#' @param x (`teal_data`) object
+#' @return (`FilteredData`) object
+#' @keywords internal
+teal_data_to_filtered_data <- function(x) { #     nolint
+  checkmate::assert_class(x, "teal_data")
+  datanames <- x@datanames
+
+  teal.slice::init_filtered_data(
+    x = as.list(x@env)[datanames],
+    join_keys = x@join_keys,
+    code = teal.data:::CodeClass$new(
+      code = paste(teal.code::get_code(x), collapse = "\n"),
+      dataname = teal.data::get_dataname(x)
+    ),
+    check = FALSE
+  )
+}
```

Comment:
I would still prefer to have this implementation in the `teal.slice` package along with the other `FilterData-utils` methods. This way we can keep on using the `teal.slice::init_filtered_data` without deprecating it, the only deprecation will be that it will stop supporting the `TealData` class.

Reply:
Maybe I overlooked something

Reply:
I imagine this will be outright replaced by `init_filtered_data`, which by then will only accept `teal_data` and have no other methods. However, as long the `TealData` class exists instantiating `FilteredData` on those objects has to be supported and then all methods of `init_filtered_data` must be somewhat consistent.

Reply:
I have an opinion that `teal.slice` should limit dependency on other `teal*` classes. `init_filtered_data` should be a generic function which needs only `data`, `join_keys`. Other stuff like `code`, `metadata`, `datalabel` etc. are not needed. Constructor could look as simple as this:

```r
init_filtered_data(
  data = list(adsl = adsl, adtte = adtte),
  join_keys = join_keys() # could be empty
)
```

`join_keys` are now `JoinKeys` (R6) class. But it could be easily represented as list. So JoinKeys could be simplified and `teal.slice` would not depend on any `teal` package.

Reply:
Thank you both for the perspectives. I agree.